### PR TITLE
Check already taken vaccination while adding child

### PR DESCRIPTION
**Pull request by laxmankhanal**

### CODE REVIEW

File: Vaccination/ViewController/ViewController.swift [hunk 1]
```diff
@@ -203,11 +166,11 @@ class ViewController: UIViewController {
 
     @IBAction func addChildForm(sender: UIGestureRecognizer) {
         if !isChildFormExpanded {
-            animateForm(345)
+            animateForm(350)
```
Comment (anmol-lf) [inline Vaccination/ViewController/ViewController.swift]: put numbers and values in sensible variables and use them

File: Vaccination/VaccineCheckListTVC.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import UIKit
+
+class VaccineCheckListTVC: UITableViewController {
+
+    @IBOutlet var vaccineCheckListTV: UITableView!
+    @IBOutlet var vaccineTableView: UITableView!
```
Comment (bishalg) [inline Vaccination/VaccineCheckListTVC.swift]: one UITableViewController can have one one UITableView if we need more then one TabelView then we need the parent viewcontroller to to be UIViewController

File: Vaccination/VaccineCheckListTVC.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    var isVaccineChecked = false
+    var i = 0
```
Comment (bishalg) [inline Vaccination/VaccineCheckListTVC.swift]: get better name for this variable

File: Vaccination/VaccineCheckListTVC.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return vaccinationList.count
+    }
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        guard let checkListCell = tableView.dequeueReusableCellWithIdentifier("vaccineCheckListCell") as? VaccineCheckListCell else {
+            return UITableViewCell() }
+        checkListCell.vaccineNameLabel.text = vaccinationList[indexPath.row].name
+
+        if  allVacc[indexPath.row]["isChecked"] == "true" {
```
Comment (bishalg) [inline Vaccination/VaccineCheckListTVC.swift]: for "BOOL" we dont need to check for true, we can simply change to 
```
if  allVacc[indexPath.row]["isChecked"]  {

File: Vaccination/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,479 @@
+import UIKit
+
+class ViewController: UIViewController {
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: get a better name for this ViewController

File: Vaccination/ViewController.swift [hunk 6]
```diff
@@ -0,0 +1,479 @@
...
+            return tableData.count
+        } else {
+            return 3
+        }
+    }
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let gender = Gender(rawValue: indexPath.row)
+        if tableView == vaccinationListTVC {
+            let cell = tableView.dequeueReusableCellWithIdentifier("vaccinationCell") as? VaccinationCell
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: use if let for cell

File: Vaccination/ViewController.swift [hunk 7]
```diff
@@ -0,0 +1,479 @@
...
+            cell!.dateLabel.text = self.tableData[indexPath.row]["date"]
+            cell!.monthLabel.text = self.tableData[indexPath.row]["month"]
+            cell!.vaccinationName.text = self.tableData[indexPath.row]["vaccinationName"]
+            cell!.timeRemaining.text = (self.tableData[indexPath.row]["date"])! + " days remaining"
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: create object rather then accessing object via dictionary

File: Vaccination/ViewController.swift [hunk 8]
```diff
@@ -0,0 +1,479 @@
...
+    }
+
+    func count() -> Int {
+        return 3
+    }
+}
+
+extension ViewController: UIScrollViewDelegate {
+       func scrollViewDidScroll(scrollView: UIScrollView) {
+                    self.view.endEditing(true)
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: correct the line spacing

File: Vaccination/ViewController.swift [hunk 9]
```diff
@@ -0,0 +1,479 @@
...
+        cell?.childImageLabel.image = UIImage(named: childImages[indexPath.row])
+        return cell!
+    }
+
+}
+
+extension ViewController: UICollectionViewDelegate {
+    func collectionView(collectionView: UICollectionView, didSelectItemAtIndexPath indexPath: NSIndexPath) {
+//        print(indexPath.row)
+//        todo display information according to the child selected
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: todo should follow the correct convention and must be the first line of comment 
```
// TODO :

File: Vaccination/ViewController/ChildViewController.swift [hunk 10]
```diff
@@ -3,3 +3,12 @@ import UIKit
 class ChildViewController: UICollectionViewController {
 
 }
+
+extension ChildViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAtIndexPath indexPath: NSIndexPath) -> CGSize {
+        let width = (self.collectionView?.frame.size.width)! / 4 - 10
```
Comment (bishalg) [inline Vaccination/ViewController/ChildViewController.swift]: add proper comment or give constant for why we are using 4 and 10 ?

File: Vaccination/ViewController/ViewController.swift [hunk 11]
```diff
@@ -288,6 +255,22 @@ class ViewController: UIViewController {
         UIApplication.sharedApplication().scheduleLocalNotification(localNotification)
     }
 
+    func clearTextFields() {
+        firstNameField.text = nil
+        lastNameField.text = nil
+        firstNameField.attributedPlaceholder = NSAttributedString(string:  "First Name", attributes:[NSForegroundColorAttributeName: UIColor.grayColor()])
+       lastNameField.attributedPlaceholder = NSAttributedString(string:  "Last Name", attributes:[NSForegroundColorAttributeName: UIColor.grayColor()])
```
Comment (bishalg) [inline Vaccination/ViewController/ViewController.swift]: fix line spacing for the line ad also fix space after string:

File: Vaccination/VaccineCheckListTVC.swift (hunk 3 above)
Comment (laxmankhanal) [inline Vaccination/VaccineCheckListTVC.swift]: :ok:

File: Vaccination/VaccineCheckListTVC.swift (hunk 2 above)
Comment (laxmankhanal) [inline Vaccination/VaccineCheckListTVC.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 5 above)
Comment (laxmankhanal) [inline Vaccination/ViewController.swift]: Sanjay have already change the name.

File: Vaccination/ViewController.swift (hunk 6 above)
Comment (laxmankhanal) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 8 above)
Comment (laxmankhanal) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 9 above)
Comment (laxmankhanal) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController/ViewController.swift (hunk 11 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/ViewController.swift]: :ok:

File: Vaccination/VaccineCheckListTVC.swift (hunk 4 above)
Comment (laxmankhanal) [inline Vaccination/VaccineCheckListTVC.swift]: :ok:

File: Vaccination/ViewController/ChildViewController.swift (hunk 10 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/ChildViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 7 above)
Comment (laxmankhanal) [inline Vaccination/ViewController.swift]: :ok:

**Comment by sanjaymhj** [comment]
Merged with notifications

**Pull request closed without merging**